File: BackEnd/ELearningAPI/ELearningAPI/TMSBO/Model/UserLesson.cs
using Library;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
namespace TMSBO.Model
{
    [ConfigTable("user_lesson", "", false, "")]
    public class UserLesson : BaseModel
    {
        [Key]
        /// <summary>
        /// user ID
        /// </summary>
        public int ID { get; set; }
        /// <summary>
        /// 
        /// </summary>
        public int? UserID { get; set; }
        /// <summary>
        /// tên user
        /// </summary>
        public string UserName { get; set; }
        /// <summary>
        /// bài học id
        /// </summary>
        public int? LessonID { get; set; }
        /// <summary>
        /// tên bài học user đang học
        /// </summary>
        public string LessonName { get; set; }
        /// <summary>
        /// 
        /// </summary>
        public int? LessonStatus { get; set; }
        /// <summary>
        /// ngày bắt đầu học
        /// </summary>
        public string StartDate { get; set; }
        /// <summary>
        /// thời gian hoàn thành
        /// </summary>
        public string FinishDate { get; set; }
        /// <summary>
        /// 
        /// </summary>
        public int? TennantID { get; set; }
        /// <summary>
        /// trạng thái 1: chưa được học, 2 đang học, 3 hoàn thành
        /// </summary>
        public int? Status { get; set; }
    }
}

File: FrontEnd/client-app/src/app/shared/model/User_Lesson.ts
import { BaseEntity } from 'src/app/base/base-model/base-entity';

/**
* Model cho user_lesson để lưu lên server
 * @export
* @class EventForInsert
*/
export class UserLesson extends BaseEntity{
    // user ID
    ID: number;
    // 
    UserID?: number;
    // tên user
    UserName: string;
    // bài học id
    LessonID?: number;
    // tên bài học user đang học
    LessonName: string;
    // 
    LessonStatus?: number;
    // ngày bắt đầu học
    StartDate: string;
    // thời gian hoàn thành
    FinishDate: string;
    // 
    TennantID?: number;
    // trạng thái 1: chưa được học, 2 đang học, 3 hoàn thành
    Status?: number;
}
